app/feature/feature-changeaddress/src/main/kotlin/com/hedvig/android/feature/changeaddress/ui/MovingDateButton.kt

```diff
@@ -0,0 +1,166 @@
+package com.hedvig.android.feature.changeaddress.ui
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.rounded.Warning
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.DatePickerDialog
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.hedvig.android.core.designsystem.component.card.HedvigCard
+import com.hedvig.android.core.designsystem.component.datepicker.HedvigDatePicker
+import com.hedvig.android.core.designsystem.material3.onWarningContainer
+import com.hedvig.android.core.designsystem.material3.warningContainer
+import com.hedvig.android.core.designsystem.material3.warningElement
+import com.hedvig.android.feature.changeaddress.ChangeAddressUiState
+import hedvig.resources.R
+import kotlinx.datetime.Instant
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toLocalDateTime
+
+@Composable
+internal fun MovingDateButton(
+  onDateSelected: (LocalDate) -> Unit,
+  uiState: ChangeAddressUiState,
```

Comment:
Could this function perhaps take in `DatePickerUiState` and `movingDate` and all the things it needs from inside the `uiState` individually, so we don't need to do the `requireNotNull` here? I think it uses a small subset of the entire `ChangeAddressUiState` object anyway, it would probably be good to only take those anyway, besides doing it so that the nullability of datePickerUiState to be enforced when calling this function.